### ISSUE TITLE
result formData  not create

### ISSUE THREAD

**Issue by brusand** [formly-js/angular-formly#150]
Hi,
Thx for yout for your component, i discover it with loopback.
i try to add a new form but it always fails (i made many tries)

here is my controler :

```js
$scope.formActorData = {};
  $scope.formFields = [{
    key: 'actorid',
    type: 'text',
    label: 'ActorID',
    required: true
  }
   ,{
    key: 'description',
    type: 'text',
    label: 'Description',
    required: true
  }];

  $scope.formOptions = {
      uniqueFormId: 'myFormActorId'

  };

  $scope.onSubmit = function() {
    console.log("Actor create " + $scope.formActorData.actorid);
    Actor.upsert($scope.formActorData, function() {
      toasty.pop.success({title: 'Actor saved', msg: 'Your actor is safe with us!', sound: false});
      $state.go('^.list');
    }, function(err) {
      console.log(err);
    });
  };
```

here is my view : 

```html
<div class="row">
  <div class="col-md-4">
    <div class="alert alert-danger" ng-if="addError">Could not add actor!</div>
    <formly-form result="formActorData" fields="formFields" options="formOptions">
        <button ng-click="onSubmit()">Hello World</button>
    </formly-form>
  </div>
</div>
<div class="controls">
    <button class="btn btn-primary btn-success" ng-click="onSubmit()">Submit</button>
</div>
```

and when i click nothing append, formActorData never updated with the new value (the form work well)

Any idea ?

thx 

best regards
Bruno

**Comment by kentcdodds** [comment]
What version of angular-formly are you using? The current version of master has significant changes from the released version. I recommend using the latest version in master as we'll be releasing this very soon and it comes with a lot of improvements.

Note, that in the latest (master) version, the attribute has changed from `result` to `model` and the `options` attribute has been removed.

Just by looking at your code, it appears you're doing everything else just fine. If making those changes doesn't fix your issue, could you please provide an example by cloning [this jsbin template](http://jsbin.com/biqesi/edit) (File -> Clone)? Thanks!

**Comment by brusand** [comment]
i use the latest version, change result to model and no change. i try your jsbin ... but you don t use the same var than me. i use $scope.formData and you (vm = this) vm.FormData.

i tried <formly-form model="$scope.formActorData" but failed too.
  .controller('ActorsCtrl', function($scope, $state, $stateParams, toasty, Actor, SweetAlert) {

  $scope.formActorData = {};
  $scope.formFields = [{
    key: 'actorid',
    type: 'text',
    label: 'ActorID',
    required: true
  }
   ,{
    key: 'description',
    type: 'text',
    label: 'Description',
    required: true
  }];

  $scope.formOptions = {
      uniqueFormId: 'myFormActorId'

  };

  $scope.onSubmit = function() {
    console.log("Actor create " + $scope.formActorData.actorid);
    Actor.upsert($scope.formActorData, function() {
      toasty.pop.success({title: 'Actor saved', msg: 'Your actor is safe with us!', sound: false});
      $state.go('^.list');
    }, function(err) {
      console.log(err);
    });
  };

  });

**Comment by brusand** [comment]
very, very strange, even with this controller, the vm.formData never be updated by the form

 
  var vm = this;

  vm.formActorData = {};
  vm.formFields = [{
    key: 'actorid',
    type: 'text',
    label: 'ActorID',
    required: true
  }
   ,{
    key: 'description',
    type: 'text',
    label: 'Description',
    required: true
  }];

  $scope.formOptions = {
      uniqueFormId: 'myFormActorId'

  };
vm.onSubmit = $scope.onSubmit;
  $scope.onSubmit = function() {
    console.log("Actor create " + vm.formActorData.actorid);
    }, function(err) {
      console.log(err);
    });
  };

  });

**Comment by brusand** [comment]
Hi Kent,

Now it works like a charm, (thanks to your link). The version of Angular is import, it failed under 1.3.8. You must include angular-formly-template too . Here is the minimal one page wich work like a charm 

<!doctype html>
<html>
<head>
    <meta charset="utf-8">
    <title>formly</title>
    <meta name="description" content="">
    <meta name="viewport" content="width=device-width">

  <script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/angularjs/1.3.8/angular.min.js"></script>
  <script src="//rawgit.com/formly-js/angular-formly/master/dist/formly.js"></script>
  <script src="//rawgit.com/formly-js/angular-formly-templates-vanilla/master/dist/angular-formly-templates-vanilla.js"></script>

  <script>
   'use strict';

/**
 * @ngdoc overview
 * @name loopbackApp
 * @description
 * # loopbackApp
 *
 * Main module of the application.
 */
angular.module('loopbackApp', ['formly'])
.controller('ActorsCtrl', function($scope) {

  $scope.actor = {};

  $scope.formFields = [{
    key: 'actorid',
    type: 'text',
    label: 'ActorID',
    required: true
  }
   ,{
    key: 'description',
    type: 'text',
    label: 'Description',
    required: true
  }];

    $scope.onSubmit = function() {
      alert("Actor create " + $scope.actor.description);
    };

  });


  </script>

</head>
<body ng-app="loopbackApp" ng-controller="ActorsCtrl">
  <div class="row">
    <div class="col-md-4">
      <div class="alert alert-danger" ng-if="addError">Could not add actor!</div>
      <formly-form model="actor" fields="formFields"  ng-submit="onSubmit()"></formly-form>
    </div>
  </div>
  <div class="controls">
    <button class="btn btn-primary btn-success" ng-click="onSubmit()">Submit</button>
  </div>
</body>
</html>


my 2 cents
Bruno

**Comment by kentcdodds** [comment]
Great! I'm glad you got it working. Let us know if you need any more help. Thanks!
